web/lib/queries/common/pageContentFields.ts

```diff
@@ -180,6 +180,36 @@ const pageContentFields = /* groq */ `
       },
     },
   },
+  _type == "promoTextTileArray"=>{
```

Comment:
better to split into another file, would also recommend splitting the action object and reuse it